organization_membership.go

```diff
@@ -0,0 +1,177 @@
+)
+
+// Compile-time proof of interface implementation.
+var _ OrganizationMemberships = (*organizationMemberships)(nil)
+
+// OrganizationMemberships describes all the organization membership related methods that
+// the Terraform Enterprise API supports.
+//
+// TFE API docs:
+// https://www.terraform.io/docs/cloud/api/organization-memberships.html
+type OrganizationMemberships interface {
+	// List all the organization memberships of the given organization.
+	List(ctx context.Context, organization string, options OrganizationMembershipListOptions) (*OrganizationMembershipList, error)
+
+	// Create a new organization membership with the given options.
+	Create(ctx context.Context, organization string, options OrganizationMembershipCreateOptions) (*OrganizationMembership, error)
+
+	// Read an organization membership by ID
+	Read(ctx context.Context, organizationMembershipID string) (*OrganizationMembership, error)
+
+	// Read an organization membership by ID with options
+	ReadWithOptions(ctx context.Context, organizationMembershipID string, options OrganizationMembershipReadOptions) (*OrganizationMembership, error)
+
+	// Delete an organization membership by its ID.
+	Delete(ctx context.Context, organizationMembershipID string) error
+}
+
+// organizationMemberships implements OrganizationMemberships.
+type organizationMemberships struct {
+	client *Client
+}
+
+// OrganizationMembershipStatus represents an organization membership status.
+type OrganizationMembershipStatus string
+
+// List all available organization membership statuses.
+const (
+	OrganizationMembershipActive  = "active"
+	OrganizationMembershipInvited = "invited"
+)
+
+// OrganizationMembershipList represents a list of organization memberships.
+type OrganizationMembershipList struct {
+	*Pagination
+	Items []*OrganizationMembership
+}
+
+// OrganizationMembership represents a Terraform Enterprise organization membership.
+type OrganizationMembership struct {
+	ID     string                       `jsonapi:"primary,organization-memberships"`
+	Status OrganizationMembershipStatus `jsonapi:"attr,status"`
+
+	// Relations
+	Organization *Organization `jsonapi:"relation,organization"`
+	User         *User         `jsonapi:"relation,user"`
+	Teams        []*Team       `jsonapi:"relation,teams"`
+}
+
+// OrganizationMembershipListOptions represents the options for listing organization memberships.
+type OrganizationMembershipListOptions struct {
+	ListOptions
+
+	Include string `url:"include"`
+}
+
+// List all the organization memberships of the given organization.
+func (s *organizationMemberships) List(ctx context.Context, organization string, options OrganizationMembershipListOptions) (*OrganizationMembershipList, error) {
+	if !validStringID(&organization) {
+		return nil, errors.New("invalid value for organization")
+	}
+
+	u := fmt.Sprintf("organizations/%s/organization-memberships", url.QueryEscape(organization))
+	req, err := s.client.newRequest("GET", u, &options)
```

Comment:
I'm a bit surprised that we don't specify a default `include` here, but the `OrganizationMembership` struct has relationships to `User` and `Teams`. Won't those be partially-populated structs unless the user sets `include` correctly? I don't think that's broken behaviour, but it feels potentially surprising.

I had a quick look around for precedent in the library, and all I could find was this TeamMember example: https://github.com/hashicorp/go-tfe/blob/8b656f1e4c1b48738f180e9b171907f2dbc2c3c0/team_member.go#L44-L48